src/app/components/inputs/select/index.tsx

```diff
@@ -1,81 +1,79 @@
+  const {
+    field: { onChange, onBlur },
+    fieldState: { error },
+  } = useController(controller);
   const { t } = useTranslation();
 
   const label = t(`inputs:label.${controller.name}`);
   const placeholder = t(`inputs:placeholder.${controller.name}`);
 
-  const handleSelect = (selectedValue: string) => {
-    field.onChange(selectedValue);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [selectedLabel, setSelectedLabel] = useState('');
+
+  const handleSelect = (selectedValue: string, valueLabel: string) => {
+    onChange(selectedValue);
+    setSelectedLabel(valueLabel);
+    setModalVisible(false);
   };
 
   return (
-    <FormControl
-      {...formControlProps}
-      isInvalid={fieldState.invalid}
-      isRequired={formControlProps.isRequired || !!controller.rules?.required}
-    >
-      <FormControlLabel>
-        <FormControlLabelText>{label}</FormControlLabelText>
-      </FormControlLabel>
-      <Select {...inputProps} onValueChange={handleSelect}>
-        <SelectTrigger variant="outline" size="md">
-          <SelectInput placeholder={placeholder} />
-          <SelectIcon marginLeft={3}>
-            <Icon as={ChevronDownIcon} />
-          </SelectIcon>
-        </SelectTrigger>
-        <SelectPortal>
-          <SelectBackdrop />
-          <SelectContent>
-            <SelectDragIndicatorWrapper>
-              <SelectDragIndicator />
-            </SelectDragIndicatorWrapper>
-            {options.map(option => (
-              <SelectItem
-                key={option.value}
-                label={option.label}
-                value={option.value}
-              />
-            ))}
-          </SelectContent>
-        </SelectPortal>
-      </Select>
-      <FormControlError>
-        <FormControlErrorText>{fieldState.error?.message}</FormControlErrorText>
-      </FormControlError>
-    </FormControl>
+    <>
+      <TextInput
+        error={error?.message}
+        label={label}
+        placeholder={placeholder}
+        {...restOfProps}
+        onBlur={onBlur}
+        value={selectedLabel}
+        readOnly
+        onChangeText={(val: string) =>
+          onChange(val.length === 1 ? val.trim() : val)
+        }
+        onPress={() => setModalVisible(true)}
+      />
+      <Modal
+        animationType="slide"
+        transparent={true}
+        visible={modalVisible}
+        onRequestClose={() => {
+          setModalVisible(false);
+        }}
+      >
+        <Overlay>
+          <ModalContent>
+            <FlatList
+              data={options}
+              renderItem={({ item }) => (
+                <TouchableOpacity
+                  onPress={() => handleSelect(item.value, item.label)}
+                >
+                  <Option>
+                    <Text fontSize="lg">{item.label}</Text>
+                  </Option>
+                </TouchableOpacity>
+              )}
+              keyExtractor={(item, i) => i.toString()}
```

Comment:
Could you use the `item.value`? Because we will have some issue with the index in the future.